Added .NET content

mdowst opened the pull request:
Created some content for working with .NET in PowerShell

mdowst committed:
initial .net files

StartAutomating commented on Guide/PowerShell/Assemblies/Namespace.md:
```diff
@@ -0,0 +1,21 @@
+Namespaces in .NET are similar to modules in PowerShell, in that they are used to organize different item into related collections. A namespace can be a collection of classes, delegates, enumerations, structures, modules (.NET modules not PowerShell), interfaces, and even other namespaces.
```
There's a bit of shorten up / punch up of content in here to do.

The bigger issue is canonical accuracy.

Aka.

You could declare multiple assemblies that use the same namespace.

The similarity to modules is fairly limited.

Also, I think that when we're talking "Assemblies / .NET", we're really _not_ talking PowerShell specifically.

That is, I think that these files could go up a directory level, so the URL would be something like

https://powershellguide.com/Languages/Namespaces

Related to this is the idea that namespaces aren't _just_ a CSharp thing.  They're also a Java thing, and a C++ thing.

mdowst replied:
I will work on shorting things up and giving more punch. 

However, I would argue that while we are talking "Assemblies / .NET" it is in relation to PowerShell. Hence the reference to modules as being a collection of similar items like a namespace. This helps the reader better visualize it in a context they would be used by some reading this guide. 

~~What if instead of calling it "Namespaces", we call it "Using .NET in PowerShell" or something like that? Then if we wanted to make a more in-depth topic into namespaces, beyond what someone would need to know for PowerShell, then it can go under Languages.~~ Disregard this. It was a long week and I had forgotten, I already wrote this. I can see now what you are say about putting it in the higher level folder. I'll work on that this weekend.

We could even make an new directory for using external tools in PowerShell. Other topics could include calling executables, Python scripts, COM objects, etc.

mdowst committed:
updates per recommendations in pull for issue #51

mdowst committed:
Merge branch 'PowerShell-Guide-Updates' of https://github.com/PowerShellGuides/PowerShellGuide into namespace-and-dot-net

mdowst committed:
pull from origin

mdowst commented:
Moved Assemblies and Namespaces into the Languages folders. Added sub-sections to them for anything PowerShell specific. Moved some content that was very PowerShell specific to the Guide\PowerShell\Concepts\PowerShell-NET-and-Everything.md file.

StartAutomating committed:
Update PowerShell-NET-and-Everything.md

Correcting PowerShell Version for ::new
